ffi/framebuffer_linux.lua

```diff
@@ -36,9 +36,12 @@ function framebuffer:init()
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
     -- Newer eink framebuffer (Kindle Touch, Paperwhite, Kobo)
     elseif ffi.string(finfo.id, 11) == "mxc_epdc_fb" then
-        -- it seems that finfo.smem_len is unreliable on kobo
         -- Figure out the size of the screen in bytes
         self.fb_size = vinfo.xres_virtual * vinfo.yres_virtual * vinfo.bits_per_pixel / 8
+        -- Kobo hasn't updated smem_len when the depth was changed from 16 to 32
+        if self.fb_size > finfo.smem_len then
```

Comment:
I feel like always use `smem_len` is probably the safest approach here. since it's literally the amount of framebuffer size that the kernel says you can use.

it's shitty that the kernel still reports double sized yres_virtual (around yres * 2) but only allocates a framebuffer with size around `xres * yres * 4`.